ang/civicase/ContactCaseTab.js

```diff
@@ -0,0 +1,215 @@
+      $scope.$emit('civicase::contact-record-case::refresh-cases');
+    };
+
+    /**
+     * Watcher for civicase::contact-record-list::loadmore event
+     *
+     * @params {Object} event
+     * @params {String} name of the list
+     */
+    function contactRecordListLoadmoreWatcher (event, name) {
+      var ind = _.findIndex($scope.casesList, function (caseObj) {
+        return caseObj.name === name;
+      });
+      var params = getCaseApiParams($scope.casesList[ind].filterParams, $scope.casesList[ind].page);
+
+      $scope.casesList[ind].showSpinner = true;
+      updateCase(ind, params);
+    }
+
+    /**
+     * Initiate cases
+     */
+    function getCases () {
+      var totalCountApi = [];
+
+      _.each($scope.casesList, function (item, ind) {
+        var params = getCaseApiParams(item.filterParams, item.page);
+
+        updateCase(ind, params);
+        totalCountApi.push(params.count);
+      });
+
+      getTotalCasesCount(totalCountApi);
+    }
+
+    /**
+     * Get parameters to load cases
+     *
+     * @param {object} filters
+     * @param {object} sort
+     * @param {object} page
+     *
+     * @return {array}
+     */
+    function getCaseApiParams (filter, page) {
+      var caseReturnParams = [
+        'subject', 'details', 'contact_id', 'case_type_id', 'status_id',
+        'contacts', 'start_date', 'end_date', 'is_deleted', 'activity_summary',
+        'activity_count', 'category_count', 'tag_id.name', 'tag_id.color',
+        'tag_id.description', 'tag_id.parent_id', 'related_case_ids'
+      ];
+      var returnCaseParams = {
+        sequential: 1,
+        return: caseReturnParams,
+        options: {
+          sort: 'modified_date ASC',
+          limit: page.size,
+          offset: page.size * (page.num - 1)
+        }
+      };
+      var params = {'case_type_id.is_active': 1};
+
+      return {
+        cases: ['Case', 'getcaselist', $.extend(true, returnCaseParams, filter, params)],
+        count: ['Case', 'getcount', $.extend(true, returnCaseParams, filter, params)]
+      };
+    }
+
+    /**
+     * Returns contact role
+     *
+     * @params {Object} cases
+     * @return {String} role
+     */
+    function getContactRole (caseObj) {
+      var contact = _.find(caseObj.contacts, {
+        contact_id: getSelectedContactId()
+      });
+
+      return contact ? contact.role : 'No Role Associated';
+    }
+
+    /**
+     * Fetches count of all the cases a contact have
+     */
+    function getTotalCasesCount (apiCall) {
+      var count = 0;
+
+      crmApi(apiCall).then(function (response) {
+        _.each(response, function (ind) {
+          count += ind;
+        });
+
+        $scope.totalCount = count;
+      });
+    }
+
+    /**
+     * Gets user id of the contact's page visited
+     *
+     * @return {String} user id if user is not selected then current logged in user id
+     */
+    function getSelectedContactId () {
+      var url = new URL(window.location.href);
```

Comment:
If we dont move this to the tpl, then this should be converted to an service so that it can be reused.